### PR TITLE
v1.10.0 - use all-attention

### PR DESCRIPTION
**Pull request by ClashLuke**

### CODE REVIEW

**Comment by ClashLuke** [comment]
As a future performance improvement, we could use [all-attention](https://arxiv.org/abs/1907.01470) for autoregressive dimensions instead of memory transformer.

**Comment by ClashLuke** [comment]
Current runs with the baseline are surprisingly strong.
image
`shuffle-new2` is the baseline without character convolution
`all-attention-no-memory` is the naive implementation of all-attention as seen [here](https://github.com/ClashLuke/JanNet/pull/24/commits/75647f74e96ba1f1667e0abb5b736a6d982d5251)

Below is the full plot
image

**Comment by ClashLuke** [comment]
The most recent commits introduced sequence slicing as well as a utils.py, which contains lovely stuff with masked automatic casting. The great thing here is that in the model I can now pass dimension names _and_ dimensions to the slicing function, while also having automatic anonymization support and more.

**Review by XMaster96 — APPROVED** [review]

**Comment by ClashLuke** [comment]
TL;DR: Much better parameter/memory as well as faster convergence.
